we are going to fetch data from a website

In [1]:
import pandas as pd 
import requests
from bs4 import BeautifulSoup

In [2]:
requests.get('https://www.ambitionbox.com/list-of-companies?page=1')

<Response [403]>

403 means bad request , server has denied our request

In [3]:
requests.get('https://www.ambitionbox.com/list-of-companies?page=1').text

'<HTML><HEAD>\n<TITLE>Access Denied</TITLE>\n</HEAD><BODY>\n<H1>Access Denied</H1>\n \nYou don\'t have permission to access "http&#58;&#47;&#47;www&#46;ambitionbox&#46;com&#47;list&#45;of&#45;companies&#63;" on this server.<P>\nReference&#32;&#35;18&#46;d7a4d417&#46;1754077217&#46;2239f5e4\n<P>https&#58;&#47;&#47;errors&#46;edgesuite&#46;net&#47;18&#46;d7a4d417&#46;1754077217&#46;2239f5e4</P>\n</BODY>\n</HTML>\n'

request has denied it is show in text

In [ ]:
headers ={ 'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.110 Safari/537.3'}
requests.get('https://www.ambitionbox.com/list-of-companies?page=1',headers =headers).text

In [2]:
requests.get('https://plugapi.vercel.app/index')

<Response [200]>

In [3]:
requests.get('https://plugapi.vercel.app/index').text

'<!doctype html>\n<html lang="en">\n  <head>\n    <meta charset="UTF-8" />\n    <link rel="icon" type="image/svg+xml" href="/assets/logo-B_u872jY.png" />\n    <meta name="viewport" content="width=device-width, initial-scale=1.0" />\n    <title>PlugAPI</title>\n    <meta name="description" content="Simplify your projects with ready-to-use authentication and user management APIs. Register, login, view users, and delete users easily with our secure API platform.">\n    <meta name="keywords" content="API platform, authentication API, user management API, register API, login API, delete user API, get all users API, backend integration, developer tools">\n    <script type="module" crossorigin src="/assets/index-B7KKLpG-.js"></script>\n    <link rel="stylesheet" crossorigin href="/assets/index-DBEXgrUJ.css">\n  </head>\n  <body>\n    <div id="root"></div>\n  </body>\n</html>\n'